src/Checkbox.js

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import styled from 'styled-components'
+import PropTypes from 'prop-types'
+import Icon from './Icon'
+import Flex from './Flex'
+import Box from './Box'
+import theme from './theme'
+
+class Checkbox extends React.Component {
+  constructor({ value, isChecked }) {
+    super()
+    this.state = {
+      value,
+      isChecked
+    }
+  }
+
+  handleClick() {
+    const { isDisabled, name, value, checkAction } = this.props
+    const { isChecked } = this.state
+    if (isDisabled) {
+      return
+    }
+
+    checkAction(name, value, !isChecked)
+    this.setState({ isChecked: !isChecked })
+  }
+
+  render() {
+    const { name, id, isDisabled, value, size, fontSize } = this.props
+
+    const currentTheme = this.props.theme || theme
+
+    const CurrentComponent = this.state.isChecked
+      ? CheckedCheckbox
+      : StyledCheckbox
+    return (
+      <CurrentComponent
+        isChecked={this.state.isChecked}
+        isDisabled={isDisabled}
+        size={size}
+        theme={currentTheme}
+        data-test="design-system-checkbox"
+        onClick={() => this.handleClick()}
+      >
+        {this.state.isChecked ? (
+          <Icon
+            name="check"
+            size={size - 4}
+            color={currentTheme.colors.white}
+          />
+        ) : null}
+
+        <input
+          type="checkbox"
```

Comment:
We spoke about this at lunch, but I think all the base form elements should accept all the non-style-related props as "passthrough" props – e.g. letting others use this Checkbox component in exactly the same way the `<input type='checkbox'>` element works. This should be less API surface area for new people to learn, and means you wouldn't have to define new props for things like `checked`, `disabled`, `onChange`, or even styled-components' `innerRef` prop